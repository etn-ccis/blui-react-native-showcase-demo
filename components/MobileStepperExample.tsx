import React, { useCallback, useState } from 'react';
import { Text, View } from 'react-native';
import { MobileStepper, DotStepperVariant } from '@brightlayer-ui/react-native-components';
import { Button, Card, Divider, useTheme } from 'react-native-paper';
import * as Colors from '@brightlayer-ui/colors';

export const MobileStepperExample: React.FC = () => {
    const theme = useTheme();
    const totalSteps = 5;
    const [currentStep, setCurrentStep] = useState(0);
    const [mobileStepperVariant, setMobileStepperVariant] = useState<DotStepperVariant>('dots');

    const updateStep = useCallback(
        (delta: number): void => {
            setCurrentStep(currentStep + delta);
        },
        [currentStep]
    );

    const cycleStepperVariant = (): void => {
        switch (mobileStepperVariant) {
            case 'dots':
                setMobileStepperVariant('text');
                break;
            case 'text':
                setMobileStepperVariant('progress');
                break;
            default:
                setMobileStepperVariant('dots');
                break;
        }
    };

    return (
        <Card style={{ padding: 0, marginTop: 0, marginHorizontal: 10, marginBottom: 10 }}>
            <Button mode="contained" onPress={(): void => cycleStepperVariant()}>
                Cycle Variant
            </Button>
            <Divider />
            <Text style={{ margin: 32, textAlign: 'center', color: theme.colors.primary }}>
                Step #{currentStep + 1}
            </Text>
            <Divider />
            <MobileStepper
                activeStep={currentStep}
                steps={totalSteps}
                leftButton={
                    <View style={{ flex: 1 }}>
                        <Button
                            style={{ width: 100, alignSelf: 'flex-start' }}
                            disabled={currentStep === 0}
                            onPress={(): void => updateStep(-1)}
                            mode="outlined"
                        >
                            Back
                        </Button>
                    </View>
                }
                rightButton={
                    <View style={{ flex: 1 }}>
                        <Button
                            style={{ width: 100, alignSelf: 'flex-end' }}
                            disabled={currentStep === totalSteps - 1}
                            onPress={(): void => updateStep(1)}
                            mode="contained"
                        >
                            Next
                        </Button>
                    </View>
                }
                variant={mobileStepperVariant}
                activeColor={Colors.blue[500]}
            />
        </Card>
    );
};
