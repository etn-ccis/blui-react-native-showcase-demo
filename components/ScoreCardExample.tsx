import React from 'react';
import { Hero, HeroBanner, InfoListItem, ScoreCard } from '@brightlayer-ui/react-native-components';
import { useTheme } from 'react-native-paper';
import { BLUIColors } from '@brightlayer-ui/colors';

export const ScoreCardExample: React.FC = () => {
    const theme = useTheme();
    return (
        <ScoreCard
            headerTitle={'Substation 42'}
            headerSubtitle={'Normal'}
            headerInfo={'42 Devices'}
            actionItems={[{ icon: { name: 'star-outline' } }, { icon: { name: 'more-vert' }, onPress: () => {} }]}
            badgeOffset={-55}
            badge={
                <HeroBanner style={{ flex: 0, minWidth: 80, justifyContent: 'flex-end' }}>
                    <Hero
                        iconBackgroundColor={theme.colors.surface}
                        label={'Score'}
                        iconSize={48}
                        iconColor={BLUIColors.success[50]}
                        ChannelValueProps={{
                            value: 98,
                            units: '/100',
                        }}
                        icon={{ family: 'brightlayer-ui', name: 'grade_a' }}
                    />
                </HeroBanner>
            }
            actionRow={<InfoListItem chevron title={'View More'} hidePadding />}
        />
    );
};
