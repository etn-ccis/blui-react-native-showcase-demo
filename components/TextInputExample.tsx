import React from 'react';
import { View, StyleSheet } from 'react-native';
import { TextInput, useTheme, HelperText, Text } from 'react-native-paper';

export const TextFieldExample: React.FC = () => {
    const theme = useTheme();
    const [normalText, setNormalText] = React.useState('');
    const [errorText, setErrorText] = React.useState('');
    const [disabledText, setDisabledText] = React.useState('');
    const [normalFilledText, setNormalFilledText] = React.useState('Hello');
    const [errorFilledText, setErrorFilledText] = React.useState('Hello');
    const [disabledFilledText, setDisabledFilledText] = React.useState('Hello');
    const [hasError, setHasError] = React.useState(true);

    const styles = StyleSheet.create({
        container: {
            marginTop: 16,
        },
        flatInput: {
            margin: 8,
        },
        outlineInput: {
            margin: 8,
            backgroundColor: 'transparent',
        },
        thinUnderline: {
            borderBottomWidth: 0, // Adjust the thickness as needed
        },
    });

    return (
        <View>
            <View style={styles.container}>
                <Text> Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorText(value);
                        setHasError(value.length > 5);
                    }}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    value={disabledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledText(value)}
                    disabled
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalFilledText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorFilledText(value);
                        setHasError(value.length > 4);
                    }}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    value={disabledFilledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledFilledText(value)}
                    disabled
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Dense Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalText}
                    dense={true}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Dense Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorText(value);
                        setHasError(value.length > 5);
                    }}
                    dense={true}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Dense Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    value={disabledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledText(value)}
                    disabled
                    dense={true}
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Dense Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalFilledText}
                    dense={true}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalFilledText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Dense Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorFilledText(value);
                        setHasError(value.length > 4);
                    }}
                    dense={true}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Dense Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    value={disabledFilledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledFilledText(value)}
                    disabled
                    dense={true}
                    style={[styles.flatInput]}
                />
            </View>
            <View style={{ marginTop: 30 }}>
                <Text> Normal Outlined TextInput </Text>
                <TextInput
                    label="Normal Outlined TextInput"
                    mode="outlined"
                    value={normalText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    outlineColor={theme.colors.outline}
                    onChangeText={(value) => setNormalText(value)}
                    style={styles.outlineInput}
                />

                <Text> Error Outlined TextInput </Text>
                <TextInput
                    label="Error Outlined TextInput"
                    mode="outlined"
                    value={errorText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    outlineColor={theme.colors.outline}
                    onChangeText={(value) => {
                        setErrorText(value);
                        setHasError(value.length > 5);
                    }}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Outlined TextInput </Text>
                <TextInput
                    label="Disabled Outlined TextInput"
                    mode="outlined"
                    value={disabledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    outlineColor={theme.colors.outline}
                    onChangeText={(value) => setDisabledText(value)}
                    disabled
                    style={styles.outlineInput}
                />
            </View>

            <View style={styles.container}>
                <Text> Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalFilledText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorFilledText(value);
                        setHasError(value.length > 4);
                    }}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    value={disabledFilledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledFilledText(value)}
                    disabled
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Dense Outline TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalText}
                    dense={true}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Dense Outline TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorText(value);
                        setHasError(value.length > 5);
                    }}
                    dense={true}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Dense Outline TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    value={disabledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledText(value)}
                    disabled
                    dense={true}
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Dense Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalFilledText}
                    dense={true}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalFilledText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Dense Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorFilledText(value);
                        setHasError(value.length > 4);
                    }}
                    dense={true}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Dense Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    value={disabledFilledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledFilledText(value)}
                    disabled
                    dense={true}
                    style={[styles.flatInput]}
                />
            </View>
        </View>
    );
};
