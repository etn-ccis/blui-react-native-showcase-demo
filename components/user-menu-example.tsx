import { IconFamily, InfoListItemProps, UserMenu } from '@brightlayer-ui/react-native-components';
import React from 'react';
import { Avatar } from 'react-native-paper';
import * as Colors from '@brightlayer-ui/colors';

const SwapIcon: IconFamily = {
    family: 'material',
    name: 'swap-horiz',
    direction: 'ltr',
};
const InvertColorsIcon: IconFamily = {
    family: 'material',
    name: 'invert-colors',
    direction: 'ltr',
};
const CancelIcon: IconFamily = {
    family: 'material',
    name: 'cancel',
    direction: 'ltr',
};

type UserMenuExampleProps = {
    onToggleRTL: () => void;
    onToggleTheme: () => void;
};

export const UserMenuExample: React.FC<UserMenuExampleProps> = (props) => {
    const { onToggleRTL, onToggleTheme } = props;

    const menuItems: InfoListItemProps[] = [
        { title: 'Toggle RTL', icon: SwapIcon, onPress: (): void => onToggleRTL() },
        {
            title: 'Toggle Theme',
            icon: InvertColorsIcon,
            onPress: (): void => onToggleTheme(),
        },
        { title: 'Cancel', icon: CancelIcon },
    ];

    return (
        <UserMenu
            menuTitle={'John Smith'}
            menuSubtitle={'Account Manager'}
            menuItems={menuItems}
            avatar={
                <Avatar.Icon
                    icon="account-circle"
                    size={40}
                    color={Colors.blue[500]}
                    style={{ backgroundColor: Colors.blue[50] }}
                />
            }
        />
    );
};
