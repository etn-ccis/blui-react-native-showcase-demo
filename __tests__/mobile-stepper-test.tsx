import 'react-native';
import React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { MobileStepperExample } from '../components/mobile-stepper-example';
import renderer from 'react-test-renderer';

describe('Kitchen sink snapshot', () => {
    test('Renders correctly and snapShot matches', ()=>{
    const snap = renderer.create(
        <SafeAreaProvider>
        <MobileStepperExample />
        </SafeAreaProvider>
    ).toJSON();
    expect(snap).toMatchSnapshot();
})
});