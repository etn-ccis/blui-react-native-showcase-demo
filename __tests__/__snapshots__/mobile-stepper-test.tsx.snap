// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kitchen sink snapshot Renders correctly and snapShot matches 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 4,
        "elevation": 1,
        "margin": 10,
        "marginBottom": 10,
        "padding": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        collapsable={false}
        index={0}
        siblings={
          Array [
            "withTheme(Divider)",
            "withTheme(Divider)",
          ]
        }
        style={
          Object {
            "backgroundColor": "#007bc1",
            "borderColor": "",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
        total={5}
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#424e54",
                    "fontFamily": "OpenSans-Regular",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "OpenSans-SemiBold",
                      "fontWeight": "600",
                    },
                    Object {
                      "fontFamily": "OpenSans-SemiBold",
                      "fontWeight": "600",
                    },
                    Array [
                      Object {
                        "color": "#ffffff",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              Cycle Variant
            </Text>
          </View>
        </View>
      </View>
      <View
        index={1}
        siblings={
          Array [
            "withTheme(Divider)",
            "withTheme(Divider)",
          ]
        }
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
        total={5}
      />
      <Text
        allowFontScaling={true}
        index={2}
        maxFontSizeMultiplier={100}
        siblings={
          Array [
            "withTheme(Divider)",
            "withTheme(Divider)",
          ]
        }
        style={
          Array [
            Object {
              "color": "#424e54",
            },
            Object {
              "writingDirection": "ltr",
            },
            Object {},
            Object {
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            Object {},
            undefined,
            Object {
              "color": "#424e54",
              "margin": 32,
              "textAlign": "center",
            },
          ]
        }
        total={5}
      >
        Step #
        1
      </Text>
      <View
        index={3}
        siblings={
          Array [
            "withTheme(Divider)",
            "withTheme(Divider)",
          ]
        }
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "height": 0.5,
            },
            undefined,
            undefined,
          ]
        }
        total={5}
      />
      <View
        index={4}
        siblings={
          Array [
            "withTheme(Divider)",
            "withTheme(Divider)",
          ]
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": 16,
            },
            undefined,
            undefined,
          ]
        }
        total={5}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "",
                "borderColor": "rgba(161, 167, 170, 0.36)",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "elevation": 0,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "width": 100,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#424e54",
                        "fontFamily": "OpenSans-Regular",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.32)",
                          "fontFamily": "OpenSans-SemiBold",
                          "fontWeight": "600",
                        },
                        Object {
                          "fontFamily": "OpenSans-SemiBold",
                          "fontWeight": "600",
                        },
                        Array [
                          Object {
                            "color": "rgba(66, 78, 84, 0.3)",
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  Back
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#b9bfc2",
                  "borderRadius": 8,
                  "height": 8,
                  "marginHorizontal": 4,
                  "overflow": "hidden",
                  "width": 8,
                },
                undefined,
                Object {
                  "backgroundColor": "#007bc1",
                },
                undefined,
              ]
            }
            testID="blui-dot"
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#b9bfc2",
                  "borderRadius": 8,
                  "height": 8,
                  "marginHorizontal": 4,
                  "overflow": "hidden",
                  "width": 8,
                },
                undefined,
                Object {},
                Object {},
              ]
            }
            testID="blui-dot"
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#b9bfc2",
                  "borderRadius": 8,
                  "height": 8,
                  "marginHorizontal": 4,
                  "overflow": "hidden",
                  "width": 8,
                },
                undefined,
                Object {},
                Object {},
              ]
            }
            testID="blui-dot"
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#b9bfc2",
                  "borderRadius": 8,
                  "height": 8,
                  "marginHorizontal": 4,
                  "overflow": "hidden",
                  "width": 8,
                },
                undefined,
                Object {},
                Object {},
              ]
            }
            testID="blui-dot"
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#b9bfc2",
                  "borderRadius": 8,
                  "height": 8,
                  "marginHorizontal": 4,
                  "overflow": "hidden",
                  "width": 8,
                },
                undefined,
                Object {},
                Object {},
              ]
            }
            testID="blui-dot"
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#007bc1",
                "borderColor": "",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 2,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0.75,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 1.5,
                "width": 100,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#424e54",
                        "fontFamily": "OpenSans-Regular",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "OpenSans-SemiBold",
                          "fontWeight": "600",
                        },
                        Object {
                          "fontFamily": "OpenSans-SemiBold",
                          "fontWeight": "600",
                        },
                        Array [
                          Object {
                            "color": "#ffffff",
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  Next
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
